bot/exts/help_channels/_message.py

```diff
@@ -42,25 +42,28 @@
 """
 
 
-async def check_for_answer(message: discord.Message) -> None:
-    """Checks for whether new content in a help channel comes from non-claimants."""
+async def update_message_caches(message: discord.Message) -> None:
+    """Checks the source of new content in a help channel and updates the appropriate cache."""
     channel = message.channel
 
     # Confirm the channel is an in use help channel
     if is_in_category(channel, constants.Categories.help_in_use):
-        log.trace(f"Checking if #{channel} ({channel.id}) has been answered.")
-
-        # Check if there is an entry in unanswered
-        if await _caches.unanswered.contains(channel.id):
-            claimant_id = await _caches.claimants.get(channel.id)
-            if not claimant_id:
-                # The mapping for this channel doesn't exist, we can't do anything.
-                return
-
-            # Check the message did not come from the claimant
-            if claimant_id != message.author.id:
-                # Mark the channel as answered
-                await _caches.unanswered.set(channel.id, False)
+        log.trace(f"Checking if #{channel} ({channel.id}) has had a reply.")
+        # Must use a timezone-aware datetime to ensure a correct POSIX timestamp.
+        timestamp = datetime.now(timezone.utc).timestamp()
```

Comment:
Need to be careful mixing aware and naïve datetimes. They all need to be one or the other. I don't remember if it was for help channels that I chose to use aware datetimes somewhere. In any case, I don't believe it's strictly necessary as long as all the code is "on the same page" and using all naïve or all aware objects. Since it's dealing with times coming from discord.py, which gives naïve objects, the path of least resistance is likely to use naïve objects everywhere.

If you swich to naïve datetimes then you should just use the time from the `message` object over `datetime.now`.

Reply:
Resolved with https://github.com/python-discord/bot/commit/9f576bcf28aa228b6c7f7d57072bba3135047200